File: 49-group-anagrams/49-group-anagrams.rb
# @param {String[]} strs
# @return {String[][]}
def group_anagrams(strs)
    groups = Hash.new { |h, k| h[k] = [] }
    strs.each do |s|
        keyStr = s.bytes
        keyStr.sort!
        keyStr = keyStr.inspect
        groups[keyStr] << s
    end
    return groups.values
end

File: 49-group-anagrams/49-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = {}
        for s in strs:
            key = [0] * 26
            for ch in s:
                key[ord(ch) - ord('a')] += 1
            key = str(key)

            groups[key] = groups.get(key, [])
            groups[key].append(s)
        return groups.values()